core/src/platform.h

```diff
@@ -33,3 +38,6 @@ std::string stringFromResource(const char* _path);
  * allocated file
  */ 
 unsigned char* bytesFromResource(const char* _path, unsigned int* _size);
+
+bool fetchData(std::unique_ptr<std::string> _url, std::stringstream& _rawData);
```

Comment:
Hmm!.

We can do streamFromURL as it is string in case of json source and binary data in case of protobuf.
But I feel fetchData is still fine, because this is making the actual http request to fetch the desired data (string or binary)